frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateCluster/CreateCluster.tsx

```diff
@@ -148,23 +152,9 @@ export default function CreateCluster(props: { infrastructureType: ClusterInfras
     (control: any) => {
       if (control.id === 'connection') {
         if (newSecret && control.setActive) {
-          const secretName = newSecret?.metadata.name ?? ''
-          if (control.providerId === 'kubevirt') {
-            // preset replacement fields to get around delayed control state from setAvailableConnections
-            control.availableMap[secretName] = {
-              replacements: {
-                pullSecret: newSecret.data?.pullSecret ?? '',
-                'ssh-publickey': newSecret.data?.['ssh-publickey'] ?? '',
-                encoded: true,
-              },
-            }
-          }
-          control.setActive(secretName)
+          control.setActive(newSecret.metadata.name)
```

Comment:
```suggestion
          const secretName = newSecret.metadata.name!
          if (control.available?.includes(secretName)) {
            control.setActive(newSecret.metadata.name)
            setNewSecret(undefined) // override with the new secret once
          }
```

I found a pre-existing bug when I was testing this. When you create a new credential in the wizard, the YAML does not get updated until you make some other change in the form. (This affects the AWS wizard as well, for example.) With this change, we wait to select the credential until it has been loaded into the dropdown as a new option (it shows up in `control.available`) and then it will be processed correctly when we select it in the dropdown.